client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import React, { FunctionComponent, createRef, useState, useEffect } from 'react';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '../../stores/onboard';
+import VerticalSelect from './vertical-select';
+import SiteTitle from './site-title';
+import { Step, usePath } from '../../path';
+import Link from '../../components/link';
+
+import './style.scss';
+
+const AcquireIntent: FunctionComponent = () => {
+	const { __: NO__ } = useI18n();
+	const { siteVertical, siteTitle } = useSelect( select => select( STORE_KEY ).getState() );
+	const makePath = usePath();
+	const siteTitleRef = createRef< HTMLInputElement >();
+
+	const [ siteTitleActive, setSiteTitleActive ] = useState( false );
+
+	useEffect( () => {
+		if ( siteTitleActive ) {
+			siteTitleRef.current?.focus();
+		}
+	}, [ siteTitleActive, siteTitleRef ] );
+
+	const handleVerticalSubmit = () => {
+		setSiteTitleActive( true );
```

Comment:
It seems `siteTitleActive` isn't use anywhere else, maybe just call `siteTitleRef.current?.focus();` here and remove the hook?

Reply:
By triggering `.focus()` without waiting for the next update cycle of the component, some related checks will produce unexpected results. 

For example, when handling blur event, currently [isFocused](https://github.com/Automattic/wp-calypso/pull/40442/files#diff-9b6c8fd6bc3641722d6cf6d98a3279fdR123) is `false` when we click on a suggestion or press Enter and `true` only when clicking outside or tabbing. By triggering the focus in the same cycle, this becomes always `true` resulting in the selection being ignored.

I don't say this should be the standard implementation but in this case is necessary until we simplify stateful logic in child component (removing `isFocused` would be a possible first step)